src/state/AppSettings.tsx

```diff
@@ -0,0 +1,26 @@
+/**
+ * Global State properties, for now just theme.
+ */
+
+export interface SettingsStateProps {
+    theme: string
+}
+
+/**
+ * App Settings class
+ */
+export class AppSettings implements SettingsStateProps {
+    theme: string
+
+    constructor(settings: SettingsStateProps) {
+        this.theme = settings.theme
+    }
+
+    /**
+     * method to create a new `AppSettings` object from an existing one
+     * with changed values
+     */
+    public copyWith(newSettings: Partial<SettingsStateProps>): AppSettings {
+        return new AppSettings({ theme: newSettings.theme ?? this.theme })
+    }
+}
```

Comment:
bah sinceramente non so se farei un'intera classe per le impostazioni, non si presta molto a essere object oriented, è semplicemente un oggetto.
nel modo in cui l'hai impostato poi con l'aggiunta di impostazioni ognuna deve essere scritta a mano sia nel costruttore che nella funzione `copyWith`, ci sono metodi più banali

se le impostazioni sono un banalissimo oggetto, lo si può copiare cambiando un parametro in modo stupido tipo
```ts
setSettings({ ...settings, theme: newTheme })
```
o cose del genere, allo stesso modo le impostazioni di default possono essere assegnate con lo stesso metodo usando lo spread operator (o assegnando un valore di default allo state dato che le impostazioni alla fine sono nello state).

giusto per fare un esempio però metti se le impostazioni non fossero uno state per qualche ragione potresti fare una cosa tipo
```ts
const settingsJSON = await AsyncStorage.getItem("settings")
const readSettings = settingsJSON ? JSON.parse(settingsJSON) : {}
settings = { ...defaultSettings, ...readSettings }
```

Ovviamente è utile definire un interface per descrivere le impostazioni, ma un'intera classe diventa tedioso e non da grossi benefici

Reply:
Ok giusto, così è molto più snello e semplice.